### PR TITLE
Add Codensity delimited continuations

### PR DESCRIPTION
**Pull request by alexfmpe**
Ported from [Control-Monad-Trans-Cont](https://hackage.haskell.org/package/transformers-0.5.6.2/docs/Control-Monad-Trans-Cont.html#g:4)
Unlike `callCC`, these seem to work even with the universal quantifier.
Curiously, it looks like similar functions work with `Yoneda`/`lowerYoneda`, though I'm not sure what those could be called.

### CODE REVIEW

**Comment by alexfmpe** [comment]
Bump

File: src/Control/Monad/Codensity.hs [hunk 1]
```diff
@@ -36,6 +36,8 @@ module Control.Monad.Codensity
   , codensityToComposedRep, composedRepToCodensity
   , wrapCodensity
   , improve
+  , reset
```
Comment (RyanGlScott) [inline src/Control/Monad/Codensity.hs:39]: Perhaps include a

```
-- ** Delimited continuations
```

Haddock section header above this line for symmetry with `Control.Monad.Trans.Cont`, which does something similar. The Haddocks for `shift`/`reset` draw upon the terminology of delimited continuations, after all, which sets them apart from other functions in this module, which describe things purely in terms of category theory. That's fine, but it's worth having a distinct section for these.

**Review by RyanGlScott — COMMENTED** [review]
Sorry for taking so long to review this.

File: src/Control/Monad/Codensity.hs [hunk 2]
```diff
@@ -283,3 +285,21 @@ improve m = lowerCodensity m
...
+--
+reset :: Monad m => Codensity m a -> Codensity m a
+reset = lift . lowerCodensity
+
+-- | @'shift' f@ captures the continuation up to the nearest enclosing
+-- 'reset' and passes it to @f@:
+--
+-- * @'reset' ('shift' f >>= k) = 'reset' (f ('lowerCodensity' . k))@
+#if __GLASGOW_HASKELL__ >= 710
+shift :: Applicative m => (forall b. (a -> m b) -> Codensity m b) -> Codensity m a
```
[the remark this answers is not in the document]
Comment (alexfmpe) [reply src/Control/Monad/Codensity.hs]: Actually me too, I just defaulted to the other style since the rest of the module did it that way

**Comment by RyanGlScott** [comment]
I've uploaded [`kan-extensions-5.2.3`](https://hackage.haskell.org/package/kan-extensions-5.2.3) to Hackage with these changes.

**Pull request merged**